python/src/spark_rapids_ml/umap.py

```diff
@@ -946,45 +966,110 @@ def _fit(self, dataset: DataFrame) -> "UMAPModel":
+            """
+            if isinstance(input, np.ndarray):
+                total_bytes = input.nbytes
+                if total_bytes <= BROADCAST_LIMIT:
+                    return [sc.broadcast(input)]
+
+                rows_per_chunk = BROADCAST_LIMIT // (total_bytes // input.shape[0])
+                num_chunks = (input.shape[0] + rows_per_chunk - 1) // rows_per_chunk
+                return [
+                    sc.broadcast(input[i * rows_per_chunk : (i + 1) * rows_per_chunk])
+                    for i in range(num_chunks)
+                ]
+
+            elif isinstance(input, dict):
+                total_bytes = sum(map(lambda arr: arr.nbytes, input.values()))
+                if total_bytes <= BROADCAST_LIMIT:
+                    return {k: [sc.broadcast(arr)] for k, arr in input.items()}
+
+                rows_per_chunk = BROADCAST_LIMIT // (
+                    total_bytes // input["data"].shape[0]
+                )
+                num_chunks = (
+                    input["data"].shape[0] + rows_per_chunk - 1
+                ) // rows_per_chunk
+                # indptr is chunked differently because it has less rows; split to match num_chunks
+                indptr_rows_per_chunk = (
+                    input["indptr"].shape[0] + (num_chunks - 1)
+                ) // num_chunks
+                return {
+                    "indices": [
+                        sc.broadcast(
+                            input["indices"][
+                                i * rows_per_chunk : (i + 1) * rows_per_chunk
+                            ]
+                        )
+                        for i in range(num_chunks)
+                    ],
+                    "indptr": [
+                        sc.broadcast(
+                            input["indptr"][
+                                i
+                                * indptr_rows_per_chunk : (i + 1)
+                                * indptr_rows_per_chunk
+                            ]
+                        )
+                        for i in range(num_chunks)
+                    ],
+                    "data": [
+                        sc.broadcast(
+                            input["data"][i * rows_per_chunk : (i + 1) * rows_per_chunk]
+                        )
+                        for i in range(num_chunks)
+                    ],
+                }  # NOTE: CSR chunks are not independently meaningful; do not use until recombined
 
-            return chunks
+            else:
+                # should never get here
+                raise ValueError(f"Unsupported input type: {type(input)}")
 
         spark = _get_spark_session()
-        broadcast_embeddings = [
-            spark.sparkContext.broadcast(chunk) for chunk in _chunk_arr(embeddings)
-        ]
-        broadcast_raw_data = [
-            spark.sparkContext.broadcast(chunk) for chunk in _chunk_arr(raw_data)
-        ]
+        broadcast_embeddings = _chunk_and_broadcast(
```

Comment:
I know this was from before, but it doesn't feel right to have the attributes of the model be broadcast objects vs just the underlying objects.  The broadcast wrapper is then applied in those cases where actual broadcast takes place, like just before transform.

Reply:
Whether to store attributes as chunked or not is not completely clear, but storing as broadcast seems off.